ratpack-core/src/test/groovy/ratpack/http/client/HttpClientSmokeSpec.groovy

```diff
@@ -291,6 +292,27 @@ class HttpClientSmokeSpec extends HttpClientSpec {
     response.statusCode == 500
   }
 
+  def "can set connect timeout"() {
+    setup:
+    def nonRoutableIp = '192.168.0.0'
+
+    when:
+    handlers {
+      get { HttpClient httpClient ->
+        httpClient.get("http://$nonRoutableIp".toURI()) {
+          it.connectTimeout(Duration.ofMillis(20))
+        } onError {
+          render it.class.name
```

Comment:
Yeah can do. I took inspiration from the read timeout test. I'm new to Ratpack, Groovy and Spock so you'll have to bear with me.